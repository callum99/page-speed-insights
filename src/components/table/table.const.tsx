import {currentPageType, historicPagesType} from "../global/global.types";

export const singlePageData:currentPageType = {
    performance: {
        TTFB: "0.1",
        FullLoadTime: "0.1",
        DOMContentLoadedTime: "0.1"
    }
};

export const historicDataFive:historicPagesType = {
    ["2024-04-24T10:52:20.779Z"]: {
        ...singlePageData
    },
    ["2024-04-24T10:12:09.664Z"]: {
        ...singlePageData
    },
    ["2024-04-24T09:34:38.019Z"]: {
        ...singlePageData
    },
    ["2024-04-24T09:10:46.905Z"]: {
        ...singlePageData
    },
    ["2024-04-24T08:13:14.099Z"]: {
        ...singlePageData
    }
};

export const historicDataFour:historicPagesType = {
    ["2024-04-24T10:52:20.779Z"]: {
        ...singlePageData
    },
    ["2024-04-24T10:12:09.664Z"]: {
        ...singlePageData
    },
    ["2024-04-24T09:34:38.019Z"]: {
        ...singlePageData
    },
    ["2024-04-24T09:10:46.905Z"]: {
        ...singlePageData
    }
};

export const historicDataThree:historicPagesType = {
    ["2024-04-24T10:52:20.779Z"]: {
        ...singlePageData
    },
    ["2024-04-24T10:12:09.664Z"]: {
        ...singlePageData
    },
    ["2024-04-24T09:34:38.019Z"]: {
        ...singlePageData
    }
};

export const historicDataTwo:historicPagesType = {
    ["2024-04-24T10:52:20.779Z"]: {
        ...singlePageData
    },
    ["2024-04-24T10:12:09.664Z"]: {
        ...singlePageData
    }
};

export const historicDataOne:historicPagesType = {
    ["2024-04-24T10:52:20.779Z"]: {
        ...singlePageData
    }
};

