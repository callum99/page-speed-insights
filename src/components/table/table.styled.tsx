import styled from 'styled-components';

export const TableContainer = styled.table`
    padding: 0 20px 20px 20px;
    width: 100%;
    border-spacing: 0;
    border-radius: 16px;
`;

export const TableHeaderStyle = styled.thead`
    & > tr > td {
        padding: 10px 5px;
        border: none;
        font-weight: bold;
        color: #ffffff;
        font-size: 12px;
        background-color: #555555;
        text-align:center;
        border-spacing: 0;
        
        &:first-child {
            border-radius: 14px 0 0 0;
        }
        
        &:last-child {
            border-radius: 0 14px 0 0;
        }
    }
`;

export const TableCellStyle= styled.td`
    color: #999999;
    background-color: #ffffff;
    padding: 10px 5px;
    border-spacing: 0;
    font-size:10px;
    text-align:center;
`;

export const TableRowStyle = styled.tr`
    &:last-child {
        & > td {
            &:first-child {
                border-radius: 0 0 0 14px;
            }
    
            &:last-child {
                border-radius: 0 0 14px 0;
            }
        }
    }
`;
