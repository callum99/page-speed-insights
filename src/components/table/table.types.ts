import {historicPagesType} from "../global/global.types";

export interface TableProps {
    data: historicPagesType;
}
