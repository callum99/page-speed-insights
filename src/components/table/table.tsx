import React from 'react';
import {TableContainer, TableHeaderStyle, TableCellStyle, TableRowStyle} from "./table.styled";
import {TableProps} from "./table.types";

const Table = ({ data }:TableProps ) => {
    const dataLength = Object.keys(data).length;
    const fifthToLastItem = dataLength - 5;
    const lessThanFiveItems = 0;
    let historicData:TableProps['data'] = {};
    let rowReturnMax;

    if (dataLength < 5) {
        rowReturnMax = lessThanFiveItems;
    } else {
        rowReturnMax = fifthToLastItem;
    }

    for(let i = rowReturnMax; i < dataLength; i++) {
        Object.assign(historicData, {[Object.keys(data)[i]]: data[Object.keys(data)[i]]});
    }

    return (
        <TableContainer role="table" aria-label="Historic Performance">
            <TableHeaderStyle role="rowgroup">
                <tr role="row">
                    <td role="columnheader">Date</td>
                    <td role="columnheader">DOMContentLoadTime</td>
                    <td role="columnheader">FullLoadTime</td>
                    <td role="columnheader">TTFB</td>
                </tr>
            </TableHeaderStyle>
            <tbody role="rowgroup">
                {historicData && Object.entries(historicData).map((page) => (
                        <TableRowStyle role="row" key={'row' + page}>
                            <TableCellStyle role="cell">{page[0]}</TableCellStyle>
                            <TableCellStyle role="cell">{page[1].performance.DOMContentLoadedTime}</TableCellStyle>
                            <TableCellStyle role="cell">{page[1].performance.FullLoadTime}</TableCellStyle>
                            <TableCellStyle role="cell">{page[1].performance.TTFB}</TableCellStyle>
                        </TableRowStyle>
                    )
                )}
            </tbody>
        </TableContainer>
    );
}

export default Table;