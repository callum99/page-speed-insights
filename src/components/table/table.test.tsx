import renderer from 'react-test-renderer';
import Table from "./table";
import {historicDataFive, historicDataFour, historicDataOne, historicDataThree, historicDataTwo} from "./table.const";

it('renders component with required props (showing max 5 rows)', () => {
    const tree = renderer.create(<Table data={historicDataFive} />).toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders component with required props (showing 4 rows)', () => {
    const tree = renderer.create(<Table data={historicDataFour} />).toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders component with required props (showing 3 rows)', () => {
    const tree = renderer.create(<Table data={historicDataThree} />).toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders component with required props (showing 2 rows)', () => {
    const tree = renderer.create(<Table data={historicDataTwo} />).toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders component with required props (showing 1 rows)', () => {
    const tree = renderer.create(<Table data={historicDataOne} />).toJSON();
    expect(tree).toMatchSnapshot();
});


it('does not render component, when no empty data object', () => {
   const tree = renderer.create(<Table  data={{}}/>).toJSON();
   expect(tree).toMatchSnapshot();
});