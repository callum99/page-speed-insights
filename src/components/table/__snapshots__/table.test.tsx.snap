// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not render component, when no empty data object 1`] = `
<table
  aria-label="Historic Performance"
  className="sc-beySPh jxDvqx"
  role="table"
>
  <thead
    className="sc-guDLey bzasjS"
    role="rowgroup"
  >
    <tr
      role="row"
    >
      <td
        role="columnheader"
      >
        Date
      </td>
      <td
        role="columnheader"
      >
        DOMContentLoadTime
      </td>
      <td
        role="columnheader"
      >
        FullLoadTime
      </td>
      <td
        role="columnheader"
      >
        TTFB
      </td>
    </tr>
  </thead>
  <tbody
    role="rowgroup"
  />
</table>
`;

exports[`renders component with required props (showing 1 rows) 1`] = `
<table
  aria-label="Historic Performance"
  className="sc-beySPh jxDvqx"
  role="table"
>
  <thead
    className="sc-guDLey bzasjS"
    role="rowgroup"
  >
    <tr
      role="row"
    >
      <td
        role="columnheader"
      >
        Date
      </td>
      <td
        role="columnheader"
      >
        DOMContentLoadTime
      </td>
      <td
        role="columnheader"
      >
        FullLoadTime
      </td>
      <td
        role="columnheader"
      >
        TTFB
      </td>
    </tr>
  </thead>
  <tbody
    role="rowgroup"
  >
    <tr
      className="sc-hLQSwg kRXjPu"
      role="row"
    >
      <td
        className="sc-dmyCSP eJrouF"
        role="cell"
      >
        2024-04-24T10:52:20.779Z
      </td>
      <td
        className="sc-dmyCSP eJrouF"
        role="cell"
      >
        0.1
      </td>
      <td
        className="sc-dmyCSP eJrouF"
        role="cell"
      >
        0.1
      </td>
      <td
        className="sc-dmyCSP eJrouF"
        role="cell"
      >
        0.1
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`renders component with required props (showing 2 rows) 1`] = `
<table
  aria-label="Historic Performance"
  className="sc-beySPh jxDvqx"
  role="table"
>
  <thead
    className="sc-guDLey bzasjS"
    role="rowgroup"
  >
    <tr
      role="row"
    >
      <td
        role="columnheader"
      >
        Date
      </td>
      <td
        role="columnheader"
      >
        DOMContentLoadTime
      </td>
      <td
        role="columnheader"
      >
        FullLoadTime
      </td>
      <td
        role="columnheader"
      >
        TTFB
      </td>
    </tr>
  </thead>
  <tbody
    role="rowgroup"
  >
    <tr
      className="sc-hLQSwg kRXjPu"
      role="row"
    >
      <td
        className="sc-dmyCSP eJrouF"
        role="cell"
      >
        2024-04-24T10:52:20.779Z
      </td>
      <td
        className="sc-dmyCSP eJrouF"
        role="cell"
      >
        0.1
      </td>
      <td
        className="sc-dmyCSP eJrouF"
        role="cell"
      >
        0.1
      </td>
      <td
        className="sc-dmyCSP eJrouF"
        role="cell"
      >
        0.1
      </td>
    </tr>
    <tr
      className="sc-hLQSwg kRXjPu"
      role="row"
    >
      <td
        className="sc-dmyCSP eJrouF"
        role="cell"
      >
        2024-04-24T10:12:09.664Z
      </td>
      <td
        className="sc-dmyCSP eJrouF"
        role="cell"
      >
        0.1
      </td>
      <td
        className="sc-dmyCSP eJrouF"
        role="cell"
      >
        0.1
      </td>
      <td
        className="sc-dmyCSP eJrouF"
        role="cell"
      >
        0.1
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`renders component with required props (showing 3 rows) 1`] = `
<table
  aria-label="Historic Performance"
  className="sc-beySPh jxDvqx"
  role="table"
>
  <thead
    className="sc-guDLey bzasjS"
    role="rowgroup"
  >
    <tr
      role="row"
    >
      <td
        role="columnheader"
      >
        Date
      </td>
      <td
        role="columnheader"
      >
        DOMContentLoadTime
      </td>
      <td
        role="columnheader"
      >
        FullLoadTime
      </td>
      <td
        role="columnheader"
      >
        TTFB
      </td>
    </tr>
  </thead>
  <tbody
    role="rowgroup"
  >
    <tr
      className="sc-hLQSwg kRXjPu"
      role="row"
    >
      <td
        className="sc-dmyCSP eJrouF"
        role="cell"
      >
        2024-04-24T10:52:20.779Z
      </td>
      <td
        className="sc-dmyCSP eJrouF"
        role="cell"
      >
        0.1
      </td>
      <td
        className="sc-dmyCSP eJrouF"
        role="cell"
      >
        0.1
      </td>
      <td
        className="sc-dmyCSP eJrouF"
        role="cell"
      >
        0.1
      </td>
    </tr>
    <tr
      className="sc-hLQSwg kRXjPu"
      role="row"
    >
      <td
        className="sc-dmyCSP eJrouF"
        role="cell"
      >
        2024-04-24T10:12:09.664Z
      </td>
      <td
        className="sc-dmyCSP eJrouF"
        role="cell"
      >
        0.1
      </td>
      <td
        className="sc-dmyCSP eJrouF"
        role="cell"
      >
        0.1
      </td>
      <td
        className="sc-dmyCSP eJrouF"
        role="cell"
      >
        0.1
      </td>
    </tr>
    <tr
      className="sc-hLQSwg kRXjPu"
      role="row"
    >
      <td
        className="sc-dmyCSP eJrouF"
        role="cell"
      >
        2024-04-24T09:34:38.019Z
      </td>
      <td
        className="sc-dmyCSP eJrouF"
        role="cell"
      >
        0.1
      </td>
      <td
        className="sc-dmyCSP eJrouF"
        role="cell"
      >
        0.1
      </td>
      <td
        className="sc-dmyCSP eJrouF"
        role="cell"
      >
        0.1
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`renders component with required props (showing 4 rows) 1`] = `
<table
  aria-label="Historic Performance"
  className="sc-beySPh jxDvqx"
  role="table"
>
  <thead
    className="sc-guDLey bzasjS"
    role="rowgroup"
  >
    <tr
      role="row"
    >
      <td
        role="columnheader"
      >
        Date
      </td>
      <td
        role="columnheader"
      >
        DOMContentLoadTime
      </td>
      <td
        role="columnheader"
      >
        FullLoadTime
      </td>
      <td
        role="columnheader"
      >
        TTFB
      </td>
    </tr>
  </thead>
  <tbody
    role="rowgroup"
  >
    <tr
      className="sc-hLQSwg kRXjPu"
      role="row"
    >
      <td
        className="sc-dmyCSP eJrouF"
        role="cell"
      >
        2024-04-24T10:52:20.779Z
      </td>
      <td
        className="sc-dmyCSP eJrouF"
        role="cell"
      >
        0.1
      </td>
      <td
        className="sc-dmyCSP eJrouF"
        role="cell"
      >
        0.1
      </td>
      <td
        className="sc-dmyCSP eJrouF"
        role="cell"
      >
        0.1
      </td>
    </tr>
    <tr
      className="sc-hLQSwg kRXjPu"
      role="row"
    >
      <td
        className="sc-dmyCSP eJrouF"
        role="cell"
      >
        2024-04-24T10:12:09.664Z
      </td>
      <td
        className="sc-dmyCSP eJrouF"
        role="cell"
      >
        0.1
      </td>
      <td
        className="sc-dmyCSP eJrouF"
        role="cell"
      >
        0.1
      </td>
      <td
        className="sc-dmyCSP eJrouF"
        role="cell"
      >
        0.1
      </td>
    </tr>
    <tr
      className="sc-hLQSwg kRXjPu"
      role="row"
    >
      <td
        className="sc-dmyCSP eJrouF"
        role="cell"
      >
        2024-04-24T09:34:38.019Z
      </td>
      <td
        className="sc-dmyCSP eJrouF"
        role="cell"
      >
        0.1
      </td>
      <td
        className="sc-dmyCSP eJrouF"
        role="cell"
      >
        0.1
      </td>
      <td
        className="sc-dmyCSP eJrouF"
        role="cell"
      >
        0.1
      </td>
    </tr>
    <tr
      className="sc-hLQSwg kRXjPu"
      role="row"
    >
      <td
        className="sc-dmyCSP eJrouF"
        role="cell"
      >
        2024-04-24T09:10:46.905Z
      </td>
      <td
        className="sc-dmyCSP eJrouF"
        role="cell"
      >
        0.1
      </td>
      <td
        className="sc-dmyCSP eJrouF"
        role="cell"
      >
        0.1
      </td>
      <td
        className="sc-dmyCSP eJrouF"
        role="cell"
      >
        0.1
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`renders component with required props (showing max 5 rows) 1`] = `
<table
  aria-label="Historic Performance"
  className="sc-beySPh jxDvqx"
  role="table"
>
  <thead
    className="sc-guDLey bzasjS"
    role="rowgroup"
  >
    <tr
      role="row"
    >
      <td
        role="columnheader"
      >
        Date
      </td>
      <td
        role="columnheader"
      >
        DOMContentLoadTime
      </td>
      <td
        role="columnheader"
      >
        FullLoadTime
      </td>
      <td
        role="columnheader"
      >
        TTFB
      </td>
    </tr>
  </thead>
  <tbody
    role="rowgroup"
  >
    <tr
      className="sc-hLQSwg kRXjPu"
      role="row"
    >
      <td
        className="sc-dmyCSP eJrouF"
        role="cell"
      >
        2024-04-24T10:52:20.779Z
      </td>
      <td
        className="sc-dmyCSP eJrouF"
        role="cell"
      >
        0.1
      </td>
      <td
        className="sc-dmyCSP eJrouF"
        role="cell"
      >
        0.1
      </td>
      <td
        className="sc-dmyCSP eJrouF"
        role="cell"
      >
        0.1
      </td>
    </tr>
    <tr
      className="sc-hLQSwg kRXjPu"
      role="row"
    >
      <td
        className="sc-dmyCSP eJrouF"
        role="cell"
      >
        2024-04-24T10:12:09.664Z
      </td>
      <td
        className="sc-dmyCSP eJrouF"
        role="cell"
      >
        0.1
      </td>
      <td
        className="sc-dmyCSP eJrouF"
        role="cell"
      >
        0.1
      </td>
      <td
        className="sc-dmyCSP eJrouF"
        role="cell"
      >
        0.1
      </td>
    </tr>
    <tr
      className="sc-hLQSwg kRXjPu"
      role="row"
    >
      <td
        className="sc-dmyCSP eJrouF"
        role="cell"
      >
        2024-04-24T09:34:38.019Z
      </td>
      <td
        className="sc-dmyCSP eJrouF"
        role="cell"
      >
        0.1
      </td>
      <td
        className="sc-dmyCSP eJrouF"
        role="cell"
      >
        0.1
      </td>
      <td
        className="sc-dmyCSP eJrouF"
        role="cell"
      >
        0.1
      </td>
    </tr>
    <tr
      className="sc-hLQSwg kRXjPu"
      role="row"
    >
      <td
        className="sc-dmyCSP eJrouF"
        role="cell"
      >
        2024-04-24T09:10:46.905Z
      </td>
      <td
        className="sc-dmyCSP eJrouF"
        role="cell"
      >
        0.1
      </td>
      <td
        className="sc-dmyCSP eJrouF"
        role="cell"
      >
        0.1
      </td>
      <td
        className="sc-dmyCSP eJrouF"
        role="cell"
      >
        0.1
      </td>
    </tr>
    <tr
      className="sc-hLQSwg kRXjPu"
      role="row"
    >
      <td
        className="sc-dmyCSP eJrouF"
        role="cell"
      >
        2024-04-24T08:13:14.099Z
      </td>
      <td
        className="sc-dmyCSP eJrouF"
        role="cell"
      >
        0.1
      </td>
      <td
        className="sc-dmyCSP eJrouF"
        role="cell"
      >
        0.1
      </td>
      <td
        className="sc-dmyCSP eJrouF"
        role="cell"
      >
        0.1
      </td>
    </tr>
  </tbody>
</table>
`;
